### ISSUE TITLE
App bar scrolling doesn't follow material spec

### ISSUE THREAD

**Issue by batbrain9392** [trimox/angular-mdc-web#998]
### What Angular MDC version are you using?
v0.35.7

### What browser(s) is this bug affecting?
Google Chrome

### What OS are you using?
Windows 10

### What are the steps to reproduce the bug?
Add an app-bar in your page or open [https://stackblitz.com/edit/angular-mdc-app-bar](https://stackblitz.com/edit/angular-mdc-app-bar)

### What is the expected behavior?
Expected - [https://material.io/design/components/app-bars-top.html#behavior](https://material.io/design/components/app-bars-top.html#behavior)
1. Upon scrolling, the top app bar can remain in place, or transform in the following ways:
   - Scrolling upward hides the top app bar
   - Scrolling downward reveals the top app bar
2. When the top app bar scrolls, its elevation above other elements becomes apparent.
3. The top app bar disappears upon scrolling up, and appears upon scrolling down.
4. Top app bars can be positioned at the same elevation as content. Upon scroll, they increase elevation and let content scroll behind them.
5. When scrolling up, prominent top app bars using imagery can transform into normal top app bars. They should not return to prominent mode until the user scrolls back to the top of the page.

### What is the actual behavior?
1. If not fixed, the app bars go out of vision and there is a white area in its stead.
2. Material design specs for app-bar scrolling are not fulfilled.

### **Current behavior of prominent app bars -**
![actual](https://user-images.githubusercontent.com/16289648/40545194-97063a2c-6048-11e8-8ba1-8d4e95a515c9.PNG)

### **Expected behavior of prominent app bars -**
![expected](https://user-images.githubusercontent.com/16289648/40545196-973b93f2-6048-11e8-8f7d-b270585e69c4.PNG)

**Comment by trimox** [comment]
Investigated and thankfully it's just a problem in the example for demo-app. Working on a fix.

**Comment by trimox** [comment]
@batbrain9392 Updated our StackBlitz example, and demo-app will be updated shortly.

https://stackblitz.com/edit/angular-mdc-app-bar?embed=1&file=app/app.component.html

**Comment by batbrain9392** [comment]
When the bar is **prominent and fixed**, then its behavior should not be static, but dynamic in the following way described :
![image](https://user-images.githubusercontent.com/16289648/41199255-015a62da-6cac-11e8-90a6-ee0d523329b6.png)

**Comment by trimox** [comment]
Yep! Thanks for the update. Fixing, this problem was cause for delaying v0.36.0.

**Comment by trimox** [comment]
@batbrain9392 Fyi, prominent app bars transforming into normal app bars is a missing feature of `material-components-web`, and can be communicated upstream for a fix.
